Caw adding nest api

chriswininger opened the pull request:
An initial commit adding the newly refactored api into the module. This version leverages the nest framework

chriswininger committed:
initial nest add

MatthewGidcomb commented on api-server-nest/package.json:
```diff
@@ -0,0 +1,79 @@
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
```
Should probably change this to `"MIT"` in case we end up copying this directory to `api-server` like we did with the Nuxt refactor.

MatthewGidcomb commented on api-server-nest/src/app.module.ts:
```diff
@@ -0,0 +1,27 @@
+            autoLoadModels: true,
+            synchronize: true,
+            host: process.env.DB_HOST,
+            port: +process.env.DB_PORT,
```
Is the `+` here to trigger a cast to `Number`? If so, seems like `parseInt` would be clearer.

MatthewGidcomb commented on api-server-nest/src/current-events/current-events.controller.ts:
```diff
@@ -0,0 +1,89 @@
+import {VERSION_1_URI} from "../utils/versionts";
+import {LoggingInterceptor} from "../logging/logging.interceptor";
+
+type EmbedableModels = typeof Venue
```
Apologies for pedantry but I think this should be `EmbeddableModels`

MatthewGidcomb commented on api-server-nest/src/current-events/dto/current-event.model.ts:
```diff
@@ -0,0 +1,127 @@
+    multi_day: boolean;
+
+    @Column
+    @ApiProperty({example: 'f467e7a0-a066-11ea-aa51-cdc3fe7afefa'})
```
Should this look more like line 40?

MatthewGidcomb commented on api-server-nest/src/current-events/dto/current-event.model.ts:
```diff
@@ -0,0 +1,127 @@
+
+    @Column
+    @ApiProperty({example: 'radio-mc-radio-station'})
+    reviewed_by_org: boolean;
```
I think this is a string and not a boolean.

chriswininger committed:
got create event workingish

chriswininger committed:
continuing to buildout api and code

chriswininger committed:
working on e2e tests and generally breaking things in the process

chriswininger committed:
beginning to get e2e rained in

the first two e2e tests now work

modified response for current-events to include the wrapper
   we should look at what happens when there is an error though, this
   response will probably be different and we really don't need status,
   but I like the idea of wrapping list responses

   We probably don't need to wrap single item responses though

I want to take one more pass at just standing up only the db layer in
the e2e tests for db state manipulation but actually launching the app
as a separate process. We could at this point also take a second pass at
using testcontainers to standup the db (maybe the app too?)

chriswininger committed:
basically working

chriswininger committed:
working better

chriswininger committed:
removed arbitrary sleeps

chriswininger committed:
improving specs

the tests now run without arbitrary timeouts
the first suite passes
added filter to strip contact_organizer

TODO: ADD THIS TO EVENTS

chriswininger committed:
two tests passing

chriswininger committed:
tests passing

chriswininger committed:
much fix, less hardcode

chriswininger committed:
remove

chriswininger committed:
use test pem

MatthewGidcomb committed:
Add test run on PRs and merges

MatthewGidcomb committed:
Manually set port for CI tests
